### DISCUSSION TITLE
How to use lifting up state ?

### DISCUSSION THREAD

**Discussion by jk0ehn** [neuefische/questions#183] [Web]
Hey folks, get this error.  

So I don't how to handle this!? Any ideas?

Heres is my pull-request
https://github.com/jk0ehn/capstone-project/pull/7

<img width="731" alt="Bildschirmfoto 2022-09-28 um 16 31 16" src="https://user-images.githubusercontent.com/109575621/192807817-1aad1fcb-87cc-4c3a-9c24-a1e991995745.png">

<img width="717" alt="Bildschirmfoto 2022-09-28 um 16 32 40" src="https://user-images.githubusercontent.com/109575621/192809014-92e03cc1-c78b-4bdc-bcae-c13cc56b1ddd.png">

<img width="584" alt="Bildschirmfoto 2022-09-28 um 16 32 58" src="https://user-images.githubusercontent.com/109575621/192809038-4cc3c0f0-7852-4d9c-aae1-32a07ae344e9.png">

**Comment by pixelass** [comment]
You have to put your state at a higher level. It has to be the parent of `Searchbar` and `ExpertCards`.


```jsx
export default function Search() {
	const [search, setSearch] = useState('');
	return (
		<>
			<Head>
				<title key="title">Search</title>
				<meta key="description" name="description" content="About" />
			</Head>
			<Searchbar search={search} onSearch={(event) => {
				setSearch(event.target.value)
			}}/>
			<ExpertCard search={search}/>
		</>
	);
}
```

Now your `Searchbar` component has to handle the `onSearch` event:

```jsx
export default function Searchbar({search, onSearch}) {

	return (
		<Wrapper>
			<MySearchbar>
				<SearchRoundedIcon color="action" />
				<MyInput
					type={search}
					onChange={onSearch}
					placeholder="Enter your location here"
				/>
			</MySearchbar>
		</Wrapper>
	);
}
```

And your `ExpertCards` needs to access `search` from the props

```js
export default function ExpertCards({search}) {
	//…
}
```

**Reply by jk0ehn** [reply]
Hey Gregor, thanks a lot. That should work. I also changed the naming of the props. Just have a look of my new pull-request.

https://github.com/jk0ehn/capstone-project/pull/7
